src/main/java/com/ibm/watson/modelmesh/ModelMesh.java

```diff
@@ -916,12 +918,34 @@ private static Metrics setUpMetrics() throws Exception {
                 params.put(kv[0], kv[1]);
             }
         }
+        String customMetricConfig = getStringParameter(MMESH_CUSTOM_ENV_VAR, null);
+        Map<String, String> customParams;
+        if (customMetricConfig == null) {
+            logger.info("{} returned null", MMESH_CUSTOM_ENV_VAR);
+            customParams = Collections.emptyMap();
+        } else {
+            logger.info("{} set to \"{}\"", MMESH_CUSTOM_ENV_VAR, customMetricConfig);
+            Matcher customMetricMatcher = CUSTOM_METRIC_CONFIG_PATT.matcher(customMetricConfig);
+            if (!customMetricMatcher.matches()) {
+                throw new Exception("Invalid metrics configuration provided in env var " + MMESH_CUSTOM_ENV_VAR + ": \""
+                                    + customMetricConfig + "\"");
+            }
+            String name = customMetricMatcher.group(1);
+            String customParamString = customMetricMatcher.group(2);
+
+            customParams = new HashMap<>();
+            customParams.put("metric_name", name);
+            for (String customParm : customParamString.substring(0).split(",")) {
+                String[] kv = customParm.split("=");
+                customParams.put(kv[0], kv[1]);
```

Comment:
Per the original proposal, the parsed values are actually names of other environment variables that should be resolved... so you should up value of env var with name `kv[1]` here.

We can throw an exception with an appropriate message if any of the specified env vars aren't set.